sycl/source/detail/device_impl.cpp

```diff
@@ -324,19 +324,32 @@ ur_native_handle_t device_impl::getNative() const {
 // clock drift between host and device.
 //
 uint64_t device_impl::getCurrentDeviceTime() {
-  using namespace std::chrono;
-  uint64_t HostTime =
-      duration_cast<nanoseconds>(steady_clock::now().time_since_epoch())
-          .count();
+  auto GetGlobalTimestamps = [this](ur_device_handle_t Device,
+                                    uint64_t *DeviceTime, uint64_t *HostTime) {
+    auto Result =
+        getAdapter()->call_nocheck<UrApiKind::urDeviceGetGlobalTimestamps>(
+            Device, DeviceTime, HostTime);
+    if (Result == UR_RESULT_ERROR_INVALID_OPERATION) {
+      // NOTE(UR port): Removed the call to GetLastError because  we shouldn't
+      // be calling it after ERROR_INVALID_OPERATION: there is no
+      // adapter-specific error.
+      throw detail::set_ur_error(
+          sycl::exception(
+              make_error_code(errc::feature_not_supported),
+              "Device and/or backend does not support querying timestamp."),
+          UR_RESULT_ERROR_INVALID_OPERATION);
```

Comment:
hm, UR should probably be returning `UR_RESULT_ERROR_UNSUPPORTED_FEATURE` instead. No need to fix this here, but maybe create an issue?

Reply:
Created https://github.com/intel/llvm/issues/18863, as you suggested I can fix that separately.